File: aizu/alds1_11_C.cpp
#include <iostream>
#include <queue>

using namespace std;
const int NIL = -1;

int n;
int A[101][101];
int d[101];

void bfs(int s) {
  queue<int> q;
  q.push(s);
  d[s] = 0;

  int u;
  while(!q.empty()) {
    u = q.front(); q.pop();
    for (int v = 0; v < n; ++v) {
      if (A[u][v] == 1 && d[v] == NIL) {
        d[v] = d[u] + 1;
        q.push(v);
      }
    }
  }
}

int main() {
  
  cin >> n;
  
  for (int i = 0; i < n; ++i) {
    for (int j = 0; j < n; ++j) {
      A[i][j] = 0;
    }
  }
  for (int i = 0; i < n; ++i) {
    d[i] = NIL;
  }

  int u, v;
  int k;
  for (int i = 0; i < n; ++i) {
    cin >> u >> k;
    for (int j = 0; j < k; ++j) {
      cin >> v;
      A[u-1][v-1] = 1;
    }
  }

  bfs(0);

  for (int i = 0; i < n; ++i) {    
    cout << i+1 << " " << d[i] << endl;
  }
}

File: aizu/alds1_11_C.py
from queue import Queue

n = int(input())
A = [[0 for i in range(n)] for j in range(n)]
d = [-1 for i in range(n)]

for i in range(n):
    u, k, *v = list(map(int, input().split()))
    for j in v:
        A[int(u)-1][int(j)-1] = 1

q = Queue()
q.put(0)
d[0] = 0

while not q.empty():
    u = q.get()
    # print('visited:', u)
    for v in range(n):
        # print(u, v)
        if A[u][v] == 1 and d[v] == -1:
            q.put(v)
            d[v] = d[u] + 1

# print('-========')
for i in range(n):
    print(i+1, d[i])